h2o-genmodel-extensions/xgboost/src/main/java/hex/genmodel/algos/xgboost/XGBoostNativeMojoModel.java

```diff
@@ -59,10 +61,16 @@ private static Booster makeBooster(byte[] boosterBytes) {
     try {
       dmat = new DMatrix(floats,1, floats.length, _sparse ? 0 : Float.NaN);
       final DMatrix row = dmat;
+      final int treeLimit;
+      if ("dart".equals(_boosterType)) {
```

Comment:
Instead of doing the decision at predict time, I would just do it before this function is called.

It would make it a little bit more obvious in MOJO reader what is going on. WDYT?

Reply:
yes it would be, but this method is static and all the context is passed via params, as such I would require a major refactor to get this done